dist/pythonlibs/riotctrl_shell/gnrc.py

```diff
@@ -116,9 +131,10 @@ def parse(self, cmd_output):
                 m = c_rtts.match(line)
                 if m is not None:
                     self._set_rtts(res, m.groupdict())
-                    return res
-        # Unable to parse RTTs, so something went wrong.
-        return None
+        if not res:
+            return None
+        else:
+            return res
```

Comment:
This I'm not sure to understand. The docstring is never testing None is returned and the default res (when the parsing doesn't work I guess) is already set to an empty dict. So it might be easy to check an empty dict instead of None to determine the parsing failed.

Reply:
You are right. 3am thickheadedness to keep the API as it is ^^.